RELEASES.md

```diff
@@ -197,11 +201,143 @@ To create a minor release:
    - Bump the TMDefaultVersion in `version.go`
    - Bump the ABCI version number, if necessary.
      (Note that ABCI follows semver, and that ABCI versions are the only versions
-     which can change during minor releases, and only field additions are valid minor changes.)
+     which can change during patch releases, and only field additions are valid patch changes.)
 4. Open a PR with these changes that will land them back on `v0.35.x`
 5. Once this change has landed on the backport branch, make sure to pull it locally, then push a tag.
    - `git tag -a v0.35.1 -m 'Release v0.35.1'`
    - `git push origin v0.35.1`
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
-   - Remove all `R:minor` labels from the pull requests that were included in the release.
+   - Remove all `R:patch` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+## Minor Release Checklist
+
+The following set of steps are performed on all releases that increment the
+_minor_ version, e.g. v0.25 to v0.26. These steps ensure that Tendermint is
+well tested, stable, and suitable for adoption by the various diverse projects
+that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters 'Feature
+Freeze' for at least two weeks. A feature freeze means that _no_ new features
+are added to the code being prepared for release. No code changes should be made
+to the code being released that do not directly improve pressing issues of code
+quality. The following must not be merged during a feature freeze:
+
+* Refactors that are not related to specific bug fixes.
+* Dependency upgrades.
+* New test code that does not test a discovered regression.
+* New features of any kind.
+* Documentation or spec improvements that are not related to the newly developed
+code.
+
+This period directly follows the creation of the [backport
+branch](#creating-a-backport-branch). The Tendermint team instead directs all
+attention to ensuring that the existing code is stable and reliable. Broken
+tests are fixed, flakey-tests are remedied, end-to-end test failures are
+thoroughly diagnosed and all efforts of the team are aimed at improving the
+quality of the code. During this period, the upgrade harness tests are run
+repeatedly and a variety of in-house testnets are run to ensure Tendermint
+functions at the scale it will be used by application developers and node
+operators.
+
+### Nightly End-To-End Tests
```

Comment:
If possible, I would rather see "nightly" tests redefined as long-running, "continual" tests, if there is not some notably prohibitive reason they must only run once in a 24h period.

Instead of "they begin running at midnight", the "long tests" logic would look something like this:

- Was a new commit pushed to `main`?
    - If yes, are the long tests running?
        - If no, begin them from this new commit to `main`
- Did the long tests just complete?
    - If yes, is there a newer commit to `main`?
        - If yes, start the long tests again from the newest commit

Running these tests continually gives faster feedback, potentially during working hours where action can be taken immediately. Running them more often offers more chances to discover flakiness or other incorrect test behavior.

If this is all out of scope of the current PR, I'm fine revisiting this later.

Reply:
I'd recommend taking a bit of a look at our current e2e testing strategies if you haven't already. We have, generally, two sets of end-to-end tests. 

https://github.com/tendermint/tendermint/tree/v0.34.x/test/e2e

1. A simplified end-to-end test that runs one every commit within a pull request.
2. A longer set of end-to-end tests that runs once per backport branch each night.

The "nightly" tests exercise a combinatorial matrix of different features and configuration flags to make sure that various aspects of the system will work in tandem together. For example, do the "legacy" and "non-legacy" p2p systems produce blocks when used by different nodes on the same network? And, does p2p statesync work with legacy p2p nodes etc.

They run for about an hour. What you're describing sounds like running the nightlies, in effect, multiple times throughout the day when new code lands. I don't exactly have an issue with that pattern, but the tests produce a lot of data that can be tough crawl through so I'd be wary of adding a lot more noise.